### ISSUE TITLE
[BP-2062] Afterpay customer IP doesn't follow IP header from config

### ISSUE THREAD

**Issue by stefanfr** [buckaroo-it/Magento2#557]
Hi,

Within the plaza we noticed that orders placed with Afterpay have the IP address form the server instead of the customer IP. Looking in the code I noticed that customerData is built with the `getRemoteAddress` function instead of following the ip headers set in the config.

The customer data array should also follow this setting to ensure the correct IP is registered in Buckaroo Plaza

**Comment by Buckaroo-Rene** [comment]
@stefanfr ,

We created a ticket for this issue with the following reference: BP-2062
It is currently on our backlog. When we have more information about the sprint schedule or status change etc, we'll let you know.

**Comment by stefanfr** [comment]
Hi @Buckaroo-Rens @Buckaroo-Rene,

After receiving an email from Chitra I noticed my earlier post isn't clear. The issue is not that we cannot change the config to follow our specific ip headers but that the buckaroo method doesn't use this flow.

If I check the flow for Ideal I see this configuration being used in file `Gateway/Http/TransactionBuilder/AbstractTransactionBuilder.php on line 532` the function getIp is called, here this config is used `$ipHeaders = $this->configProviderAccount->getIpHeader($store);`

However when I check the Afterpay, now Riverty, flow I see that the customerData is build with only the remote address:
```
$customerData = [
        [
            '_'    => $this->getRemoteAddress(),
            'Name' => 'CustomerIPAddress',
        ],
        [
            '_'    => $accept,
            'Name' => 'Accept',
        ]
    ];
```
`$this->getRemoteAddress()` should include the `$alternativeHeaders` parameter to correctly set the customer Ip

**Comment by Buckaroo-Rens** [comment]
Thanks for your additional info @stefanfr, we will look into this asap.

**Comment by Buckaroo-Rene** [comment]
Hi @stefanfr ,

We've just released version 1.44.0 of the Buckaroo Plugin.
A fix for this issue is included [BP-2062](https://github.com/buckaroo-it/Magento2/releases/tag/v1.44.0)
